mrp_unbuild_tracked_raw_material/models/unbuild.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2019 Akretion (http://www.akretion.com). All Rights Reserved
+# @author David BEAL <david.beal@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+import logging
+
+from odoo import _, models
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as DT_FORMAT
+
+
+logger = logging.getLogger(__name__)
+INFO = "Previous write can lead to concurrency updates. " \
+       "Ensure this event is not occurs frequently in your process"
+
+
+class MrpUnbuild(models.Model):
+    _inherit = "mrp.unbuild"
+
+    def action_unbuild(self):
+        # We collect all components grouped by unbuild order
+        components_by_serial = {
+            unbuild: unbuild.bom_id.bom_line_ids.mapped(
+                "product_id").filtered(lambda s: s.tracking == "serial")
+            for unbuild in self}
+        components_by_lot = {
+            unbuild: unbuild.bom_id.bom_line_ids.mapped(
+                "product_id").filtered(lambda s: s.tracking == "lot")
+            for unbuild in self}
+        for unbuild in self:
+            components = components_by_serial[unbuild] | components_by_lot[
+                unbuild]
+            # To bypass Odoo limitation in use cases we need
+            # to make components as untracked
+            components.write({"tracking": "none"})
```

Comment:
Personally I don't like this... and you are aware of the problem as you log it in the next line.

I think if you catch the exception [here](https://github.com/odoo/odoo/blob/12.0/addons/mrp/models/mrp_unbuild.py#L115) you can continue the process with a context (something like "bypass_unbuild_tracked_products=True") and then you can inherit the create of stock.move.line searching for that context to correct the values passes [here](https://github.com/odoo/odoo/blob/12.0/addons/mrp/models/mrp_unbuild.py#L140) with the correct lot.

This way you will only be affecting this process and not potentially breaking the data consistency in parallel processes.

What do you think?

PD: I loved the description of the PR :laughing:

Reply:
Thanks for your answer @lreficent.
Ok to catch the exception, but if I do that i have a problem with
https://github.com/odoo/odoo/blob/12.0/addons/mrp/models/mrp_unbuild.py#L134
because I have no manufacturing order then `original_move` becomes False and no creation of stock.move.line.

Reply:
I see, then you don't even need the context, you can create the stock.move.lines (and any other action needed for your use case) after calling super.